### DISCUSSION TITLE
Clear cached javascript

### DISCUSSION THREAD

**Discussion by Besler** [Kitware/trame#224] [Q&A]
This might be naive, but I think my browser is caching the javascript I serve over trame. As a consequence, when I run an update, the browsers which used the old version are not getting the new version.

Is there a way to prevent this? Typically I would put the version number in the javascript file. Does that seem reasonable given the trame docker setup with Apache?

(I can disable caching with dev tools, but that's not so great for production)

**Comment by ghost** [comment]
Yes, adding a version number to the JavaScript file is a common technique to prevent browser caching. By appending a unique version number to the file URL, you force the browser to download the new version of the file, even if it's already in the cache. For an example:

```
<script src="https://example.com/scripts/main.js?v=1"></script>
```

In this example, the v=1 parameter serves as the version number. When you update the JavaScript file, you can increment the version number to force the browser to download the new version.

In terms of implementing this in your trame Docker setup with Apache, you can add a RewriteRule to your .htaccess file to append the version number to the file URL. Here's an example:

```
RewriteEngine On
RewriteRule ^scripts/main.js$ scripts/main.js?v=1 [L]
```

**Comment by jourdain** [comment]
I tried to remove any caching from trame default webapp but indeed we still run into caching issue. Is it your app or a dependency that you created that needs update? Hopefully the speed of `trame-*` should be fairly slow and the caching issue should only really happen at dev time (easy to solve) rather than production time (harder). 

If it's your app/dep, you can add that fake version in the scripts array of your Python module.
